examples/codes/run_process_example.ex.py

```diff
@@ -0,0 +1,376 @@
+    Constraint.TF_JCRIT_RATIO_UPPER_LIMIT,
+    Constraint.TF_CURRENT_DENSITY_UPPER_LIMIT,
+    Constraint.TF_T_MARGIN_LOWER_LIMIT,
+    Constraint.PSEPB_QAR_UPPER_LIMIT,
+):
+    template_builder.add_constraint(constraint)
+
+
+# %%[markdown]
+# Many of these constraints require certain iteration variables to have been
+# specified, or certain input values. The novice user can easily not be
+# aware that this is the case, or simply forget to specify these.
+
+# The :py:class:`PROCESSTemplateBuilder` will warn the user if certain
+# values have not been specified. For example, if we try to make a set of
+# inputs for an IN.DAT now, we will get many warning messages:
+
+# %%
+inputs = template_builder.make_inputs()
+
+
+# %%[markdown]
+# So let's go ahead and add the iteration variables we want to the problem:
+
+# %%
+template_builder.add_variable("bt", 5.3292, upper_bound=20.0)
+template_builder.add_variable("rmajor", 8.8901, upper_bound=13.0)
+template_builder.add_variable("te", 12.33, upper_bound=150.0)
+template_builder.add_variable("beta", 3.1421e-2)
+template_builder.add_variable("dene", 7.4321e19)
+template_builder.add_variable("q", 3.5, lower_bound=3.5)
+template_builder.add_variable("pheat", 50.0)
+template_builder.add_variable("ralpne", 6.8940e-02)
+template_builder.add_variable("bore", 2.3322, lower_bound=0.1)
+template_builder.add_variable("ohcth", 0.55242, lower_bound=0.1)
+template_builder.add_variable("thwcndut", 8.0e-3, lower_bound=8.0e-3)
+template_builder.add_variable("thkcas", 0.52465)
+template_builder.add_variable("tfcth", 1.2080)
+template_builder.add_variable("gapoh", 0.05, lower_bound=0.05, upper_bound=0.1)
+template_builder.add_variable("gapds", 0.02, lower_bound=0.02)
+template_builder.add_variable("cpttf", 6.5e4, lower_bound=6.0e4, upper_bound=9.0e4)
+template_builder.add_variable("tdmptf", 2.5829e01)
+template_builder.add_variable("fcutfsu", 0.80884, lower_bound=0.5, upper_bound=0.94)
+template_builder.add_variable("fvsbrnni", 0.39566)
+
+# %%[markdown]
+# Many of the PROCESS constraints use so-called 'f-values', which are automatically
+# added to the iteration variables using this API. However, often one wants to modify
+# the defaults of these f-values, which one can do as such:
+
+# %%
+# Modified f-values and bounds w.r.t. defaults
+template_builder.adjust_variable("fne0", 0.6, upper_bound=0.95)
+template_builder.adjust_variable("fdene", 1.2, upper_bound=1.2)
+
+
+# %%[markdown]
+# Often one wants to specify certain impurity concentrations, and even use
+# one of these as an iteration variable.
+# Note that the PROCESS term for the impurity concentrations is the `fimp`
+# array, for which sadly one must still know the integer in order to specify
+# it as an iteration variable. 13 is Xenon.
+
+# %%
+template_builder.add_impurity(Impurities.H, 1.0)
+template_builder.add_impurity(Impurities.He, 0.1)
+template_builder.add_impurity(Impurities.W, 5.0e-5)
+template_builder.add_variable("fimp(13)", 3.573e-04)
```

Comment:
Need to check it works but this, If not we could fix the string produced to add the close bracket or something. There must be a reason I didnt close the bracket originally...: 

```suggestion
template_builder.add_variable(Impurities.Xe.id(), 3.573e-04)
```

Reply:
ahh looks like in old MFILEs they hadn't added the closing bracket and in new ones they have so if the id function is fixed all of the above will work just need to close the bracket here: 
https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/bdde3af24fb31f215d9588947c273c044cc28042/bluemira/codes/process/api.py#L157